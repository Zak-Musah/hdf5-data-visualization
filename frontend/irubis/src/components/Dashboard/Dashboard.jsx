import React, { useEffect, useState } from "react";
import Plot from "react-plotly.js";
function Dashboard({ testData, testName }) {
  const [glucose, setGlucose] = useState([]);
  const [time, seTime] = useState([]);
  const [measurement, seMeasurement] = useState([]);
  console.log(testData);

  useEffect(() => {
    seTime(testData.data.glucose);
    setGlucose(testData.data.glucose);
  }, [testData]);

  return (
    <>
      {glucose && (
        <Plot
          data={{
            x: time,
            y: glucose,
            type: "date",
            mode: "line",
            name,
            line: {
              width: 3,
            },
            hoverinfo: "closest",
          }}
          // layout={}
          config={{
            responsive: true,
            useResizeHandler: true,
            displaylogo: false,
            displayModeBar: true,
          }}
        />
      )}
    </>
  );
}

export default Dashboard;
