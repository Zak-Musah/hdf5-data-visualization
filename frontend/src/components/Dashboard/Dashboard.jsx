import React, { useState, useEffect } from "react";
import { Spinner } from "react-bootstrap";
import axios from "axios";

import Chart from "../Chart/Chart";

function Dashboard(props) {
  const [fileNames, setFileNames] = useState([]);
  const [fileName, setFileName] = useState("");
  const [fileData, setFileData] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    axios
      .get(`${process.env.REACT_APP_BACKEND_SERVICE_URL}/api/get_file_names`)
      .then((response) => {
        if (response.data.status === true) {
          setFileNames(response.data.data);
        } else {
          setFileNames([]);
        }
      })
      .catch((err) => console.log(err));
  }, []);

  const fetchData = (index) => {
    const name = fileNames[index];
    setFileName(name);
    setLoading(true);
    const data = { file_name: name };
    axios
      .post(`${process.env.REACT_APP_BACKEND_SERVICE_URL}/api/read_data`, data)
      .then((response) => {
        if (response.data.status === true) {
          setFileData(response.data.data);
        } else {
          setFileData([]);
        }
        setLoading(false);
      })
      .catch((err) => console.log(err));
  };

  const clearDashboard = () => {
    setFileData([]);
    setFileName("");
  };

  return (
    <>
      <div className="col-3 bg-light card">
        <div className="card-body">
          <h5 className="card-title">Select File</h5>
          <h6 className="card-subtitle mb-2 text-muted">
            To show dataset on dashboard
          </h6>
          <div className="list-group">
            {fileNames &&
              fileNames.map((fileName, index) => (
                <button
                  type="button"
                  className="list-group-item list-group-item-action"
                  onClick={() => fetchData(index)}
                  key={index}
                >
                  {fileName.slice(0, -5).toUpperCase()}
                </button>
              ))}
            <button
              onClick={() => clearDashboard()}
              className="btn btn-primary"
              type="button"
              key="clear-dashboard"
            >
              Clear Dashboard
            </button>
          </div>
        </div>
      </div>
      <div className="col bg-light card">
        <div className="card-body" style={{ paddingLeft: "20%"}}>
          {loading && (
            <center>
              <Spinner animation="border" role="status">
                <span className="sr-only">Loading...</span>
              </Spinner>
            </center>
          )}
          {!loading && fileName.length > 0 ? (
            <>
              <h4 className="card-title">
                {fileName.slice(0, -5).toUpperCase()}
              </h4>
              <Chart fileData={fileData} fileName={fileName} />
            </>
          ) : (
            <h4 className="card-title">No data yet...</h4>
          )}
        </div>
      </div>
    </>
  );
}

export default Dashboard;
