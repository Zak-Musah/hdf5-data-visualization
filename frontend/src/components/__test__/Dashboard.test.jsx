import React from "react";
import { mount, render } from "enzyme";
import renderer from "react-test-renderer";
import axios from "axios";

import Dashboard from "../Dashboard/Dashboard";

test("Dashboard renders properly", () => {
  const wrapper = render(<Dashboard />);
  expect(wrapper.find(".card-body").length).toBe(2);
  expect(wrapper.find(".card-title").first().text()).toBe("Select File");
  expect(wrapper.text()).toMatch("No data yet...");
});

test("Dashboard renders with snapshot", () => {
  const tree = renderer.create(<Dashboard />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Dashboard get file names", async () => {
  const wrapper = mount(<Dashboard />);
  axios.get(`http://localhost:5000/api/get_file_names`).then((response) => {
    expect(wrapper).toMatch(
      `{"data":["test_dataset_1.hdf5","test_dataset_2.hdf5"],"message":"Data files retrieved","status":true}`,
    );
  });
});

test("Dashboard click on first file name", () => {
  const wrapper = mount(<Dashboard />);
  const element = wrapper.find("button").first();
  expect(element).toHaveLength(1);
  element.simulate("click");
  expect(wrapper).toMatchSnapshot();
  wrapper.unmount();
});

test("Dashboard click on clear dashboard", () => {
  const wrapper = mount(<Dashboard />);
  const element = wrapper.find("button").last();
  expect(element).toHaveLength(1);
  element.simulate("click");
  expect(wrapper).toMatchSnapshot();
  wrapper.unmount();
});
