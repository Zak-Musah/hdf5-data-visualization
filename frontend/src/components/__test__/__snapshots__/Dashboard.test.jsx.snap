// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders with snapshot 1`] = `
Array [
  <div
    className="col-3 bg-violet card"
  >
    <div
      className="card-body"
    >
      <div
        className="list-group"
      >
        <button
          className="btn btn-success"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#1e1f45",
              "transition": "box-shadow 5s",
            }
          }
          type="button"
        >
          <h3>
            Choose Dataset
          </h3>
        </button>
      </div>
      <br />
    </div>
  </div>,
  <div
    className="col bg-light card"
  >
    <div
      className="card-body"
      style={
        Object {
          "paddingLeft": "4%",
        }
      }
    >
      <h4
        className="card-title"
        style={
          Object {
            "color": "white",
          }
        }
      >
        No data yet...
      </h4>
    </div>
  </div>,
]
`;
